From 0fa3b26a792d4bc26c5c19d12812d757d7f6036c Mon Sep 17 00:00:00 2001
From: kxxt <rsworktech@outlook.com>
Date: Thu, 15 Jan 2026 13:35:33 +0100
Subject: [PATCH] Copy Chrome ffmpeg config for riscv64

For some reason, electron is using Chrome's config instead of Chromium's
---
 chromium/config/Chrome/linux/riscv64/config.h             | 8 ++++----
 .../config/Chrome/linux/riscv64/libavutil/ffversion.h     | 2 +-
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/chromium/config/Chrome/linux/riscv64/config.h b/chromium/config/Chrome/linux/riscv64/config.h
index bf936999ff..b90460fe20 100644
--- a/chromium/config/Chrome/linux/riscv64/config.h
+++ b/chromium/config/Chrome/linux/riscv64/config.h
@@ -1,12 +1,12 @@
 /* Automatically generated by configure - do not modify! */
 #ifndef FFMPEG_CONFIG_H
 #define FFMPEG_CONFIG_H
-/* #define FFMPEG_CONFIGURATION "--disable-everything --disable-all --disable-doc --disable-htmlpages --disable-manpages --disable-podpages --disable-txtpages --disable-static --enable-avcodec --enable-avformat --enable-avutil --enable-static --enable-libopus --disable-debug --disable-bzlib --disable-error-resilience --disable-iconv --disable-network --disable-schannel --disable-sdl2 --disable-symver --disable-xlib --disable-zlib --disable-securetransport --disable-faan --disable-alsa --disable-iamf --disable-autodetect --enable-decoder='vorbis,libopus,flac' --enable-decoder='pcm_u8,pcm_s16le,pcm_s24le,pcm_s32le,pcm_f32le,mp3' --enable-decoder='pcm_s16be,pcm_s24be,pcm_mulaw,pcm_alaw' --enable-demuxer='ogg,matroska,wav,flac,mp3,mov' --enable-parser='opus,vorbis,flac,mpegaudio,vp9' --extra-cflags=-I/usr/local/src/chromium/src/third_party/opus/src/include --disable-linux-perf --x86asmexe=nasm --optflags='\"-O2\"' --arch=riscv64 --extra-cflags='-march=rv64gc' --enable-cross-compile --target-os=linux --sysroot=/usr/local/src/chromium/src/build/linux/debian_trixie_riscv64-sysroot --extra-cflags='--target=riscv64-linux-gnu' --extra-ldflags='--target=riscv64-linux-gnu' --enable-pic --cc=clang --cxx=clang++ --ld=clang --extra-ldflags='-fuse-ld=lld' --enable-decoder='aac,h264' --enable-demuxer=aac --enable-parser='aac,h264'" -- elide long configuration string from binary */
+/* #define FFMPEG_CONFIGURATION "--disable-everything --disable-all --disable-doc --disable-htmlpages --disable-manpages --disable-podpages --disable-txtpages --disable-static --enable-avcodec --enable-avformat --enable-avutil --enable-static --enable-libopus --disable-debug --disable-bzlib --disable-error-resilience --disable-iconv --disable-network --disable-schannel --disable-sdl2 --disable-symver --disable-xlib --disable-zlib --disable-securetransport --disable-faan --disable-alsa --disable-iamf --disable-autodetect --enable-decoder='vorbis,libopus,flac' --enable-decoder='pcm_u8,pcm_s16le,pcm_s24le,pcm_s32le,pcm_f32le,mp3' --enable-decoder='pcm_s16be,pcm_s24be,pcm_mulaw,pcm_alaw' --enable-demuxer='ogg,matroska,wav,flac,mp3,mov' --enable-parser='opus,vorbis,flac,mpegaudio,vp9' --extra-cflags=-I/home/kxxt/electron-ci/sources/electron/src/third_party/opus/src/include --disable-linux-perf --x86asmexe=nasm --optflags='\"-O2\"' --arch=riscv64 --extra-cflags='-march=rv64gc' --enable-cross-compile --target-os=linux --sysroot=/home/kxxt/electron-ci/sources/electron/src/build/linux/debian_trixie_riscv64-sysroot --extra-cflags='--target=riscv64-linux-gnu' --extra-ldflags='--target=riscv64-linux-gnu' --enable-pic --cc=clang --cxx=clang++ --ld=clang --extra-ldflags='-fuse-ld=lld' --enable-decoder='aac,h264' --enable-demuxer=aac --enable-parser='aac,h264'" -- elide long configuration string from binary */
 #define FFMPEG_LICENSE "LGPL version 2.1 or later"
 #define CONFIG_THIS_YEAR 2025
 #define FFMPEG_DATADIR "/usr/local/share/ffmpeg"
 #define AVCONV_DATADIR "/usr/local/share/ffmpeg"
-#define CC_IDENT "clang version 22.0.0git (https://chromium.googlesource.com/a/external/github.com/llvm/llvm-project 8a5f15330feb693bc2c3923c0e77ce808382491f)"
+#define CC_IDENT "clang version 22.0.0git (https://chromium.googlesource.com/a/external/github.com/llvm/llvm-project efe9a8c95451c9dadb5dd522802b05afd8b52d1b)"
 #define OS_NAME linux
 #define EXTERN_PREFIX ""
 #define EXTERN_ASM 
@@ -382,7 +382,7 @@
 #define HAVE_INLINE_ASM_LABELS 1
 #define HAVE_INLINE_ASM_NONLOCAL_LABELS 1
 #define HAVE_PRAGMA_DEPRECATED 1
-#define HAVE_RSYNC_CONTIMEOUT 1
+#define HAVE_RSYNC_CONTIMEOUT 0
 #define HAVE_SYMVER_ASM_LABEL 1
 #define HAVE_SYMVER_GNU_ASM 1
 #define HAVE_VFP_ARGS 0
@@ -442,7 +442,7 @@
 #define HAVE_PERL 1
 #define HAVE_POD2MAN 1
 #define HAVE_TEXI2HTML 0
-#define HAVE_XMLLINT 1
+#define HAVE_XMLLINT 0
 #define HAVE_ZLIB_GZIP 0
 #define HAVE_OPENVINO2 0
 #define CONFIG_DOC 0
diff --git a/chromium/config/Chrome/linux/riscv64/libavutil/ffversion.h b/chromium/config/Chrome/linux/riscv64/libavutil/ffversion.h
index 7cf8fc496a..a1a458cd45 100644
--- a/chromium/config/Chrome/linux/riscv64/libavutil/ffversion.h
+++ b/chromium/config/Chrome/linux/riscv64/libavutil/ffversion.h
@@ -1,5 +1,5 @@
 /* Automatically generated by version.sh, do not manually edit! */
 #ifndef AVUTIL_FFVERSION_H
 #define AVUTIL_FFVERSION_H
-#define FFMPEG_VERSION "N-123075-g71b8313894"
+#define FFMPEG_VERSION "git-2026-01-15-c40aaf7fe4"
 #endif /* AVUTIL_FFVERSION_H */
-- 
2.39.2

